src/components/users.vue

```diff
@@ -1,135 +1,142 @@
-              break;
-            case 'roles':
-              leftColumn = left.applications.length;
-              rightColumn = right.applications.length;
-              break;
-            default:
-              leftColumn = left[this.sortColumn];
-              rightColumn = right[this.sortColumn];
+      filterUsers: function() {
+        this.filteredUsers = this.users.map(u => {
+          let date =  new Date(parseInt(u._id.$oid.substring(0, 8), 16) * 1000);
+          let shortDate = '' + date
+          u._id.$oid = shortDate.slice(4,15)
+          if (u.full_name.toLowerCase().includes(this.filterString.toLowerCase()) || u.email.toLowerCase().includes(this.filterString.toLowerCase())) {
+            return u;
           }
-
-          let sort = 0;
-          if (leftColumn > rightColumn) {
-            sort = 1;
-          } else if(leftColumn < rightColumn) {
-            sort = -1;
-          }
-
-          return (this.sortAscending ? sort : -sort);
-        });
-      },
-      filteredUsers: function() {
-        return this.users.filter(user => {
-          let userRoles = user.applications.map(application => { return application.type; });
-          return this.filterRole === "" || userRoles.includes(this.filterRole);
         });
       },
+      testing: function() {
+        return dateFromObjectId =  new Date(parseInt(oid.substring(0, 8), 16) * 1000);
+      }
     },
     methods: {
-      sortBy: function(currentColumn) {
-        if (this.sortColumn === currentColumn) {
-          this.sortAscending = !this.sortAscending;
-        } else {
-          this.sortAscending = defaultSortAscending;
+      sortUsers: function(column) {
+        if (column === 'Name') {
+          this.status.Name = 'fa fa-angle-down';
+          this.status.Email = 'fa fa-angle-right';
+          this.status.Registered = 'fa fa-angle-right';
+          this.status.Role = 'fa fa-angle-right';
+          this.filteredUsers = _.orderBy(this.filteredUsers, 'full_name');
+        }
+        if (column === 'Email') {
+          this.status.Name = 'fa fa-angle-right';
+          this.status.Email = 'fa fa-angle-down';
+          this.status.Registered = 'fa fa-angle-right';
+          this.status.Role = 'fa fa-angle-right';
+          this.filteredUsers = _.orderBy(this.filteredUsers, 'email');
+        }
+        if (column === 'Registered') {
+          this.status.Name = 'fa fa-angle-right';
+          this.status.Email = 'fa fa-angle-right';
+          this.status.Registered = 'fa fa-angle-down';
+          this.status.Role = 'fa fa-angle-right';
+          this.filteredUsers = this.users;
+        }
+        if (column === 'Role') {
+          this.status.Name = 'fa fa-angle-right';
+          this.status.Email = 'fa fa-angle-right';
+          this.status.Registered = 'fa fa-angle-right';
+          this.status.Role = 'fa fa-angle-down';
+          this.filteredUsers = _.orderBy(this.filteredUsers, 'role');
         }
```

Comment:
Consider using a `switch` statement here instead of multiple `if`s in a row: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/switch